opentracing/trace_context.go

```diff
@@ -0,0 +1,164 @@
+package opentracing
+
+import "sync"
+
+// A TraceContextID is the smallest amount of state needed to describe a span's
+// identity within a larger [potentially distributed] trace. The TraceContextID
+// is not intended to encode the span's operation name, timing, or log data,
+// but merely any unique identifiers (etc) needed to contextualize it within a
+// larger trace tree.
+//
+// TraceContextIDs are sufficient to propagate the, well, *context* of a particular
+// trace from process to process.
+type TraceContextID interface {
```

Comment:
one big design decision here was to separate `TraceContextID` from `TraceContext` (see below).